src/main/java/com/revlearn/team1/service/transaction/TransactionServiceImp.java

```diff
@@ -1,56 +1,76 @@
-        TransactionModel retrievedTransaction = transactionRepo.findById(transactionId).orElseThrow(
-                () -> new TransactionNotFoundException(String
-                        .format("Could not find transaction by Id in Database.  Transaction ID: %d", transactionId)));
-        return transactionMapper.toResDTO(retrievedTransaction);
-    }
+    @Value("${CLIENT_URL}")
+    private String clientUrl;
 
     @Override
-    public List<TransactionResponseDTO> getTransactions() {
-        List<TransactionModel> transactionModels = transactionRepo.findAll();
-        return transactionModels.stream()
-                .map(transactionMapper::toResDTO)
-                .collect(Collectors.toList());
-    }
+    public TransactionResponseDTO checkout(TransactionRequestDTO request) throws StripeException {
+        TransactionModel item = transactionMapper.fromDTO(request);
 
-    @Override
-    public void deleteTransactionById(int transactionId) {
-        transactionRepo.findById(transactionId).orElseThrow(
-                () -> new TransactionNotFoundException(String.format(
-                        "Could not find transaction to delete by that Id in Database.  Transaction ID: %d",
-                        transactionId)));
+        // Set the Stripe API key
+        Stripe.apiKey = stripeApiKey;
+
+        // Create a Product in Stripe (optional, can be reused or stored in DB)
+        var productParams = ProductCreateParams.builder()
+                .setName(item.getName())
+                .build();
+        var product = Product.create(productParams);
+
+        // Create a Price for the product in Stripe
+        var priceParams = PriceCreateParams.builder()
+                .setProduct(product.getId())
+                .setUnitAmount(item.getPrice()) // Price in cents (e.g., $10.00 = 1000 cents)
+                .setCurrency("usd")
+                .build();
+        var price = Price.create(priceParams);
 
-        transactionRepo.deleteById(transactionId); // if no exception, delete the id
+        // Create a checkout session
+        var sessionParams = SessionCreateParams.builder()
+                .setMode(SessionCreateParams.Mode.PAYMENT) // Payment mode
+                .addLineItem(
+                        SessionCreateParams.LineItem.builder()
+                                .setPrice(price.getId()) // Use the price created above
+                                .setQuantity(item.getQuantity()) // Quantity of items
+                                .build()
+                )
+                .setSuccessUrl(clientUrl + "/checkout-success") // Redirect on successful payment
+                .setCancelUrl(clientUrl + "/checkout-cancel") // Redirect if payment is canceled
+                .build();
+
+        // TODO: Kafka implementation here to persist the transaction data to our database
+
+        // Create and return the session URL
+        Session session = Session.create(sessionParams);
```

Comment:
IDK if this is right but here's a full implementation ChatGPT gave me of what I'm talking about

`// Set the Stripe API key
Stripe.apiKey = stripeApiKey;

// Define product data directly within PriceData
var productData = SessionCreateParams.LineItem.PriceData.ProductData.builder()
        .setName(item.getName()) // Set the product name or description
        .build();

// Define PriceData with the product information
var priceData = SessionCreateParams.LineItem.PriceData.builder()
        .setCurrency("usd")
        .setUnitAmount(item.getPrice()) // Price in cents (e.g., $10.00 = 1000 cents)
        .setProductData(productData) // Use the defined product data
        .build();

// Define LineItem with PriceData
var lineItem = SessionCreateParams.LineItem.builder()
        .setPriceData(priceData) // Use PriceData directly
        .setQuantity(item.getQuantity()) // Quantity of items
        .build();

// Define SessionCreateParams
var sessionParams = SessionCreateParams.builder()
        .setMode(SessionCreateParams.Mode.PAYMENT) // Payment mode
        .addLineItem(lineItem) // Add LineItem
        .setSuccessUrl(clientUrl + "/checkout-success") // Redirect on successful payment
        .setCancelUrl(clientUrl + "/checkout-cancel") // Redirect if payment is canceled
        .build();

// Create and return the checkout session URL
Session session = Session.create(sessionParams);

// TODO: Kafka implementation here to persist the transaction data to our database
kafkaTemplate.send("payment-session-created", session.getId(), session);

// Return the transaction response DTO
return new TransactionResponseDTO(session.getUrl(), "Payment Processed!");